File: frontend/src/api/workspaces.ts
import type { ApiClient } from "./client"

export interface WorkspaceSummaryDto {
  id: string
  name: string
  slug: string
  role: string
}

export interface CurrentUserDto {
  user_id: string
  email: string | null
  workspaces: WorkspaceSummaryDto[]
}

export async function getCurrentUser(apiClient: ApiClient): Promise<CurrentUserDto> {
  return apiClient.request<CurrentUserDto>("/api/v1/me")
}

File: backend/app/schemas/workspaces.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class WorkspaceSummary(BaseModel):
    id: UUID
    name: str
    slug: str
    role: str


class CurrentUserResponse(BaseModel):
    user_id: UUID
    email: str | None
    workspaces: list[WorkspaceSummary]


class CreateWorkspaceRequest(BaseModel):
    name: str = Field(min_length=1, max_length=120)
    slug: str = Field(pattern=r"^[a-z0-9]+(?:-[a-z0-9]+)*$")


class WorkspaceResponse(BaseModel):
    id: UUID
    name: str
    slug: str
    created_at: datetime
